packages/shopping/src/services/user-management.service.ts

```diff
@@ -83,4 +86,29 @@ export class UserManagementService implements UserService<User, Credentials> {
     await this.userRepository.userCredentials(user.id).create({password});
     return user;
   }
+
+  async requestPasswordReset(email: string): Promise<NodeMailer> {
+    const noAccountFoundError =
+      'No account associated with the provided email address.';
+    const foundUser = await this.userRepository.findOne({
+      where: {email},
+    });
+    if (!foundUser) {
+      throw new HttpErrors.NotFound(noAccountFoundError);
+    }
+    foundUser.resetKey = _.times(20, () => _.random(35).toString(36)).join('');
```

Comment:
Maybe use the following:
`import { v4 as uuidv4 } from 'uuid';`
`uuidv4();`

Continuing my anti lodash crusade. :)

Reply:
I wouldn't install an additional package for such a logic. We already have lodash being used in the project and it works well 😄

Reply:
I think you'll find 'uuid' in the package-lock, correct me if I'm wrong.

Reply:
Sorry haven't checked but you can always verify and confirm the same

Reply:
I checked and uuid is not installed and again I honestly don't see the need to install a new package for this. If perhaps concern is anti-lodash then we can use default JS `Math.random().toString(36).slice(2)`

Reply:
The change would remove the repeated random, which may cause confusion. The real reason for suggestion was that uuid v4 generated value creates random element that is not linked to timestamp. Lodash comment was really about me always trying to get rid of the expensive package.

Maybe better solution than v4 though:
https://medium.com/javascript-in-plain-english/you-might-not-need-uuid-v4-for-generating-random-identifiers-89e8a28a7d77

Maybe just use random and add comment to use something else in production?

Reply:
If you dont want to add `uuid ` I would recommend using the native `crypto.randomBytes` function directly. `uuid` uses this function internally to generate its random values because it is the best way to generate secure random values ([CSPRNG](https://gist.github.com/joepie91/7105003c3b26e65efcea63f3db82dfba)) in nodejs. Just need to ensure that the value is base64url encoded

```ts
const token = randomBytes(32)
        .toString("base64")
        .replace(/=/g, "")
        .replace(/\//g, "_")
        .replace(/\+/g, "-");
```

Reply:
> I checked and uuid is not installed

I'm not sure where you checked but 'uuid' will be installed as per the package lock file.  Double checked and it is indeed found in 'node_modules' on install.  Please reconsider using uuid v4... it is a good option for this use case.